### ISSUE TITLE
side-effects-cache does not work in gitlab CI

### ISSUE THREAD

**Issue by leaumar** [pnpm/pnpm#3201]
### Code to reproduce the issue:

.gitlab-ci.yml
```yaml
image: node:14.16.0-alpine

cache:
  paths:
    - .pnpm-store/

variables:
  npm_config_store_dir: .pnpm-store
  npm_config_side_effects_cache: "true"

ci:
  script:
    - apk add --no-cache alpine-sdk autoconf automake git libtool make nasm python util-linux
    - npm install -g pnpm
    - pnpm install --frozen-lockfile
```

pnpm-workspaces.yaml
```yaml
packages:
  - packages/*
```

(workspaces are detected correctly as shown by pnpm reporting their count during installs: 1 root + 4 modules)

packages/x/package.json
```json
    "dependencies": {
        "@project/common": "workspace:../common",
        "@project/lib": "workspace:../lib",
        "basic-auth": "2.0.1",
        "bytes-iec": "3.1.1",
        "formidable": "1.2.2",
        "forwarded-parse": "2.1.1",
        "get-stream": "6.0.0",
        "http-status-codes": "2.1.4",
        "imagemin": "7.0.1",
        "imagemin-mozjpeg": "9.0.0",
        "imagemin-pngquant": "9.0.1",
        "imagemin-webp": "6.0.0",
        "koa": "2.13.1",
        "koa-compose": "4.1.0",
        "koa-compress": "5.0.1",
        "koa-helmet": "6.0.0",
        "koa-router": "10.0.0",
        "lodash": "4.17.21",
        "mime": "2.5.2",
        "mustache": "4.1.0",
        "ow": "0.23.0",
        "pretty-ms": "7.0.1",
        "range-parser": "1.2.1",
        "resolve-path": "1.4.0",
        "stream-meter": "1.0.4",
        "ts-essentials": "7.0.1",
        "type-fest": "0.21.2"
    },
```

### Expected behavior:

1. first CI run fetches dependencies, builds binary ones locally (imagemin-*), and caches whole store including binaries
2. second CI run extracts cache, the `install` phase needs to do no heavy lifting as the full store is restored, it just needs to relink node_modules

### Actual behavior:

1. first CI run fetches dependencies, builds binary ones locally (imagemin-*), and caches whole store (about 14k files from the whole project's dependencies)
    - deleting the cache and running without `side-effects-cache` **reports the same number of files being cached as _with_ `side-effects-cache`**, so clearly nothing extra ends up in the store for caching by setting the flag
2. second CI run extracts cache, **the `install` phase compiles the binary dependencies again**

> ⚠ spawn /builds/xxx/node_modules/.pnpm/mozjpeg@7.0.0/node_modules/mozjpeg/vendor/cjpeg ENOENT
> ⚠ mozjpeg pre-build test failed
> ℹ compiling from source
> ⚠ spawn /builds/xxx/node_modules/.pnpm/cwebp-bin@5.1.0/node_modules/cwebp-bin/vendor/cwebp ENOENT
> ⚠ cwebp pre-build test failed
> ℹ compiling from source

Caching side-effects is important for me to greatly reduce build times (especially because my CI minutes are limited).

### Additional information:

 - `node -v` prints: 14.16.0
 - OS: Dockerized alpine linux

**Comment by leaumar** [comment]
Strangely, specifically installing version 5.11.1 for example does not fix the behavior, binaries are still rebuilt despite the store being cached and restored. What else could be wrong or could I be overlooking? :/

**Comment by bhovhannes** [comment]
I believe it is fixed in pnpm v7.x. Because we've observed expected behavior yesterday.

Yesterday we've disabled `side-effects-cache` in our GitLab enterprise pipeline config because postinstall tasks were not run when installation happened from cache. We are using playwright and its binary gets downloaded during postinstall stage. As soon as we started to cache pnpm store, we lose that binary because postinstall tasks were not run.
